db/database.go

```diff
@@ -945,24 +948,26 @@ func (db *Database) UpdateAllDocChannels(doCurrentDocs bool, doImportDocs bool)
 		}
 		var err error
 		if db.UseXattrs() {
-			err = db.Bucket.WriteUpdateWithXattr(key, KSyncXattrName, 0, func(currentValue []byte, currentXattr []byte, cas uint64) (raw []byte, rawXattr []byte, err error) {
+			err = db.Bucket.WriteUpdateWithXattr(key, KSyncXattrName, 0, func(currentValue []byte, currentXattr []byte, cas uint64) (raw []byte, rawXattr []byte, deleteDoc bool, err error) {
+				// There's no scenario where a doc should from non-deleted to deleted during UpdateAllDocChannels processing, so deleteDoc is always returned as false.
```

Comment:
This comment is a bit confusing .. what ensures that the doc is always going from non-deleted -> deleted in this callback?  Does that comment apply to the entire callback, or just if `currentValue` is nil or 0?